src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -903,6 +918,20 @@ public AppPage verifyHtmlMainContent(String filePath) throws IOException {
         return verifyHtmlPart(MAIN_CONTENT, filePath);
     }
 
+    public AppPage verifyHtmlMainContentWithReloadRetry(final String filePath) throws IOException {
+        return RetryManager.runUntilNoException(new RetryableTaskReturnsThrows<AppPage, IOException>("HTML verification") {
+            @Override
+            public AppPage run() throws IOException {
```

Comment:
I think the weirdness is because the `AssertionError` is written at the end (second parameter of the `runUntilNoException` method), and isn't very visible. It'd look less weird if the user saves the task into a variable `task`; the call would then look like `RetryManager.runUntilNoException(task, AssertionError)`.